runtime/lua/vim/ui/img.lua

```diff
@@ -0,0 +1,557 @@
+---Id of the last image created.
+---@type integer
+local LAST_IMAGE_ID = 0
+
+---@class vim.ui.Image
+---@field id integer unique id associated with the image
+---@field bytes string|nil bytes of the image loaded into memory
+---@field filename string path to the image on disk
+---@field private __metadata vim.ui.img.Metadata
+local M = {}
+M.__index = M
+
+---Path to the directory where images may be cached.
+---@type string
+---@diagnostic disable-next-line:param-type-mismatch
+M.cache_dir = vim.fs.joinpath(vim.fn.stdpath('cache'), 'img')
+
+---Collection of names to associated providers used to display and manipulate images.
+---@type vim.ui.img.Providers
+M.providers = require('vim.ui.img.providers')
+
+---Creates a new image instance, optionally taking pre-loaded bytes.
+---@param opts string|{bytes?:string, filename:string}
+---@return vim.ui.Image
+function M.new(opts)
+  vim.validate('opts', opts, { 'string', 'table' })
+  if type(opts) == 'table' then
+    vim.validate('opts.bytes', opts.bytes, 'string', true)
+    vim.validate('opts.filename', opts.filename, 'string')
```

Comment:
"file" is common in `vim.fs`. Eventually we will have URIs, and "file" is more compatible with that.

```suggestion
    vim.validate('opts.file', opts.file, 'string')
```